src/main/java/org/opensearch/geospatial/ip2geo/common/Ip2GeoLockService.java

```diff
@@ -43,6 +43,19 @@ public Ip2GeoLockService(final ClusterService clusterService, final Client clien
         this.lockService = new LockService(client, clusterService);
```

Comment:
Shouldn't we block this constructor if we don't want plugin to instantiate their own lock service?

Reply:
Yes, but first all usages in the default distribution need to be addressed: https://github.com/search?q=org%3Aopensearch-project%20LockService&type=code

I'll raise a PR in JS to formally mark the constructor as Deprecated

Reply:
Oh. I mean, `public Ip2GeoLockService(final ClusterService clusterService, final Client client)` not `public LockService(final ClusterService clusterService, final Client client)`

Reply:
yes, good point. The tests need to be updated in Ip2GeoLockServiceTests. Taking a look now.

Reply:
Removed the constructor of `Ip2GeoLockService` that accepts a `Client`. There are still instances where `LockService` is instantiated in tests that I plan to remove in a subsequent PR. The difficulty I am facing now is creating a mock instance of LockService because its a final class in the JS SPI.